lnrpc/routerpc/router_server.go

```diff
@@ -0,0 +1,265 @@
+	}
+
+	return routerServer, macPermissions, nil
+}
+
+// Start launches any helper goroutines required for the rpcServer to function.
+//
+// NOTE: This is part of the lnrpc.SubServer interface.
+func (s *Server) Start() error {
+	if atomic.AddInt32(&s.started, 1) != 1 {
+		return nil
+	}
+
+	return nil
+}
+
+// Stop signals any active goroutines for a graceful closure.
+//
+// NOTE: This is part of the lnrpc.SubServer interface.
+func (s *Server) Stop() error {
+	if atomic.AddInt32(&s.shutdown, 1) != 1 {
+		return nil
+	}
+
+	close(s.quit)
+
+	return nil
+}
+
+// Name returns a unique string representation of the sub-server. This can be
+// used to identify the sub-server and also de-duplicate them.
+//
+// NOTE: This is part of the lnrpc.SubServer interface.
+func (s *Server) Name() string {
+	return subServerName
+}
+
+// RegisterWithRootServer will be called by the root gRPC server to direct a
+// sub RPC server to register itself with the main gRPC root server. Until this
+// is called, each sub-server won't be able to have requests routed towards it.
+//
+// NOTE: This is part of the lnrpc.SubServer interface.
+func (s *Server) RegisterWithRootServer(grpcServer *grpc.Server) error {
+	// We make sure that we register it with the main gRPC server to ensure
+	// all our methods are routed properly.
+	RegisterRouterServer(grpcServer, s)
+
+	log.Debugf("Router RPC server successfully register with root gRPC " +
+		"server")
+
+	return nil
+}
+
+// SendPayment attempts to route a payment described by the passed
+// PaymentRequest to the final destination. If we are unable to route the
+// payment, or cannot find a route that satisfies the constraints in the
+// PaymentRequest, then an error will be returned. Otherwise, the payment
+// pre-image, along with the final route will be returned.
+func (s *Server) SendPayment(ctx context.Context,
+	req *PaymentRequest) (*PaymentResponse, error) {
+
+	switch {
+	// If the payment request isn't populated, then we won't be able to
+	// even attempt a payment.
+	case req.PayReq == "":
+		return nil, fmt.Errorf("a valid payment request MUST be specified")
+	}
+
+	// Now that we know the payment request is present, we'll attempt to
+	// decode it in order to parse out all the parameters for the route.
+	payReq, err := zpay32.Decode(req.PayReq, s.cfg.ActiveNetParams)
+	if err != nil {
+		return nil, err
+	}
+
+	// Atm, this service does not support invoices that don't have their
+	// value fully specified.
+	if payReq.MilliSat == nil {
+		return nil, fmt.Errorf("zero value invoices are not supported")
+	}
+
+	// Now that all the information we need has been parsed, we'll map this
+	// proto request into a proper request that our backing router can
+	// understand.
+	finalDelta := uint16(payReq.MinFinalCLTVExpiry())
+	payment := routing.LightningPayment{
+		Target:            payReq.Destination,
+		Amount:            *payReq.MilliSat,
+		FeeLimit:          lnwire.MilliSatoshi(req.FeeLimit),
```

Comment:
I the fee limit is in msat, I think it should also be passed into the rpc as msat? Or alternatively change the field in  `LightningPayment` and further to `btcutil.Amount`?